### PR TITLE
4313 address duplicate matching feedback

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

This addresses feedback from the duplicate matches product review

## Changes proposed in this pull request
 - Add back link to `/support/candidates/{{id}}/status`
 - Back link dynamically changes based on resolved match for the following page `support/duplicate-matches/{{id}}`
 - Add CSS classes for more than 2 digits in the blue circle on `support/duplicate-matches`
 - Fix extra space for the linter in `app/views/support_interface/candidates/edit_candidate_account_status.html.erb` 

## Guidance to review

- Enable the `duplicate_matching` feature flag
- Open the relevent pages and check the back links work as expected

- Get over 99 duplicate matches and check that you can still see the three or four digit number within the blue circle

## Link to Trello card

https://trello.com/c/OLYe4yz6/4313-address-duplicate-matching-feedback

### CODE REVIEW

File: app/frontend/styles/support/_app-count.scss [hunk 1]
```diff
@@ -2,12 +2,22 @@
...
   vertical-align: middle;
   margin-left: 3px;
 }
+
+.app-count-standard {
+  width: 26px;
+  height: 26px;
+}
+
+.app-count-large {
```
Comment (duncanjbrown) [inline app/frontend/styles/support/_app-count.scss:18]: I'm not sure where we are with BEM these days — should this be `app-count` and `app-count--large`, @frankieroberto?

**Comment by frankieroberto** [comment]
@avinhurry I've added commit 8dd40a9b9681a912438de7fedb1a59366ba6863a which simplifies the `app-count` component by allowing it to expand automatically into a lozenge shape if if hits 3 digits (hopefully rare!) whilst remaining a circle when it contains 1 or 2 digits...

Screenshot 2022-01-20 at 14 54 19

Screenshot 2022-01-20 at 14 54 33

Screenshot 2022-01-20 at 14 54 40

Let me know what you think!?

File: app/views/support_interface/candidates/edit_candidate_account_status.html.erb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+<% content_for :before_content, govuk_back_link_to(support_interface_candidate_path, 'Back') %>
```
Comment (frankieroberto) [inline app/views/support_interface/candidates/edit_candidate_account_status.html.erb:1]: How does this path magically know the ID of the candidate? 🤯

Comment (tomas-stefano) [reply app/views/support_interface/candidates/edit_candidate_account_status.html.erb:1]: It uses the referer if the url doesn't exist. I will amend the candidate to the url (or just put `:back` - I'll see).

File: app/views/support_interface/duplicate_matches/show.html.erb [hunk 3]
```diff
@@ -1,5 +1,10 @@
 <% content_for :browser_title, SupportInterface::DuplicateMatchesTableComponent.description_for(@match) %>
-<% content_for :before_content, govuk_back_link_to(support_interface_duplicate_matches_path, 'Back') %>
+
+<% if @match.resolved? %>
+  <% content_for :before_content, govuk_back_link_to(support_interface_duplicate_matches_path(resolved: true), 'Back') %>
+<% else %>
+  <% content_for :before_content, govuk_back_link_to(support_interface_duplicate_matches_path, 'Back') %>
+<% end %>
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply app/views/support_interface/duplicate_matches/show.html.erb]: Done! ♻️

File: app/views/support_interface/candidates/edit_candidate_account_status.html.erb (hunk 2 above)
Comment (tomas-stefano) [reply app/views/support_interface/candidates/edit_candidate_account_status.html.erb]: Done!

**Comment by avinhurry** [comment]
> @avinhurry I've added commit [8dd40a9](https://github.com/DFE-Digital/apply-for-teacher-training/commit/8dd40a9b9681a912438de7fedb1a59366ba6863a) which simplifies the `app-count` component by allowing it to expand automatically into a lozenge shape if if hits 3 digits (hopefully rare!) whilst remaining a circle when it contains 1 or 2 digits...
> 
> Screenshot 2022-01-20 at 14 54 19
> 
> Screenshot 2022-01-20 at 14 54 33
> 
> Screenshot 2022-01-20 at 14 54 40
> 
> Let me know what you think!?

Looks great @frankieroberto! CSS magic 🤯, how does that work?

**Comment by frankieroberto** [comment]
> Looks great @frankieroberto! CSS magic 🤯, how does that work?

`min-width` is the magic CSS property. It sets a minimum width (so the count starts as a circle) but allows it to widen if needed to fit the content.

**Pull request merged**